xray/utils/resultwriter.go

```diff
@@ -132,16 +146,22 @@ func printMessage(message string) {
 	log.Output("💬" + message)
 }
 
-func GenerateSarifFileFromScan(extendedResults *ExtendedScanResults, isMultipleRoots, markdownOutput bool, scanningTool, toolURI string) (string, error) {
-	report, err := sarif.New(sarif.Version210)
+func GenerateSarifContentFromResults(extendedResults *ExtendedScanResults, isMultipleRoots, includeLicenses, markdownOutput bool) (sarifStr string, err error) {
```

Comment:
why not inside sarifutils?

Reply:
@sverdlov93 Since there is no Sarif struct at the arguments and the returned value.
This method uses the sarif utils to accomplish that but at the end you generate an output from this method to write.
Like printing table or simple-json, I think it, and its related private methods, belongs to resultwriter